src/site/antora/modules/ROOT/pages/manual/cloud.adoc

```diff
@@ -15,549 +15,267 @@
+* *Console* can act as a proxy.
+Logs written to console can be consumed by an external service.
+For example, https://12factor.net/logs[The Twelve-Factor App] and https://kubernetes.io/docs/concepts/cluster-administration/logging/[Kubernetes Logging Architecture] recommends this approach.
 
-Similar to sending logs to a forwarder, logs can also be sent to a cluster of aggregators.
-However, setting this up is not as simple since, to be highly available, a cluster of aggregators must be used.
-However, the SocketAppender currently can only be configured with a single host and port.
-To allow  for failover if the primary aggregator fails the SocketAppender must be enclosed in a  xref:manual/appenders.adoc#FailoverAppender[FailoverAppender], which would also have the secondary aggregator configured.
-Another option is to have the SocketAppender  point to a highly available proxy that can forward to the Log Aggregator.
+* A *library* can act as proxy.
+It can tap into the logging API and forward it to an external service.
+For instance, https://docs.datadoghq.com/logs/log_collection/java[Datadog's Java Log Collector] uses this mechanism.
 
-If the log aggregator used is Apache Flume (or similar) the Appenders for these support
-being configured with a list of hosts and ports so high availability is not an issue.
+* An external *service* can act as a proxy, which applications can write logs to.
+For example, you can write to https://www.elastic.co/logstash[Logstash], a https://kubernetes.io/docs/concepts/cluster-administration/logging/#sidecar-container-with-a-logging-agent[Kubernetes logging agent sidecar], or a https://redis.io/glossary/redis-queue/[Redis queue] over a socket.
 
-image:LoggerAggregator.png[Aggregator, "Application Logging to an Aggregator via TCP"]
+What to use as a proxy depends on your deployment environment.
+You should consult to your colleagues if there is already an established logging proxy convention.
+Otherwise, we strongly encourage you to establish one in collaboration with your system administrators and architects.
 
-[#ELK]
-== Logging using Elasticsearch, Logstash, and Kibana
+[#appender]
+=== Configure your appender correctly
 
-There are various approaches with different trade-offs for ingesting logs into
-an ELK stack. Here we will briefly cover how one can forward Log4j generated
-events first to Logstash and then to Elasticsearch.
+Once you decide on <<proxy,the log proxy>> to use, the choice of appender pretty much becomes self-evident.
+Nevertheless, there are some tips we recommend you to practice:
 
-=== Log4j Configuration
+* *For writing to console*, use a xref:manual/appenders.adoc#ConsoleAppender[Console Appender] and make sure to configure its `direct` attribute to `true` for the maximum efficiency.
 
-==== JsonTemplateLayout
+* *For writing to an external service*, use a xref:manual/appenders.adoc#SocketAppender[Socket Appender] and make sure to configure the null delimiter of the associated layout.
```

Comment:
Should we mention to users to prefer `TCP` over `UDP`?
At least if `SocketAppender` is concerned. We don't have an appender that uses `QUIC`.